File: MediaLoader/src/main/java/com/app/medialoader/tinyhttpd/response/Response.java
package com.app.medialoader.tinyhttpd.response;

import com.app.medialoader.tinyhttpd.HttpVersion;
import com.app.medialoader.tinyhttpd.HttpHeaders;

import java.io.IOException;

/**
 * 响应接口
 *
 * @author vincanyang
 */
public interface Response {

    HttpStatus status();

    void setStatus(HttpStatus status);

    HttpVersion protocol();

    HttpHeaders headers();

    /**
     * 添加头部
     *
     * @param key   {@link com.app.medialoader.tinyhttpd.HttpHeaders.Names}
     * @param value
     */
    void addHeader(String key, String value);

    void write(byte[] bytes) throws IOException;

    void write(byte[] bytes, int offset, int length) throws IOException;
}

File: MediaLoader/src/main/java/com/app/medialoader/tinyhttpd/response/Response.kt
package com.app.medialoader.tinyhttpd.response

import com.app.medialoader.tinyhttpd.HttpHeaders
import com.app.medialoader.tinyhttpd.response.HttpStatus
import com.app.medialoader.tinyhttpd.HttpVersion
import java.io.IOException
import kotlin.Throws

/**
 * 响应接口
 *
 * @author vincanyang
 */
interface Response {
    fun status(): HttpStatus
    fun setStatus(status: HttpStatus)
    fun protocol(): HttpVersion
    fun headers(): HttpHeaders

    /**
     * 添加头部
     *
     * @param key   [com.app.medialoader.tinyhttpd.HttpHeaders.Names]
     * @param value
     */
    fun addHeader(key: String?, value: String?)

    @Throws(IOException::class)
    fun write(bytes: ByteArray)

    @Throws(IOException::class)
    fun write(bytes: ByteArray, offset: Int, length: Int)
}